src/Npgsql/NpgsqlWriteBuffer.cs

```diff
@@ -129,9 +130,9 @@ public async Task Flush(bool async)
 
         #region Direct write
 
-        internal async Task DirectWrite(byte[] buffer, int offset, int count, bool async)
+        internal void DirectWrite(ReadOnlySpan<byte> buffer)
```

Comment:
If we now have a fully synchronous DirectWrite, we should make the version below (which accepts ReadOnlyMemory) async-only, and not accept an async flag (rename to DirectWriteAsync).

If we want to, we can still have a DirectWrite accepting a byte[] and dispatching to either the Span or Memory version based on the async flag, so that consumers which don't care (e.g. ByteaHandler) don't have to dispatch themselves. Though I don't have strong feelings about this.

Reply:
This was my initial intent, unfortunately the `DirectWrite(ReadOnlyMemory output, bool async)` method is called in [ByteaHandler](https://github.com/manandre/npgsql/blob/cac45bc264c720aaefbf0a46b4d86c10ce68f628/src/Npgsql/TypeHandlers/ByteaHandler.cs#L127) with a `ReadOnlyMemory` instance from the `INpgsqlTypeHandler<ReadOnlyMemory<byte>>` interface.
Any idea on how to avoid/change that?